staging/src/k8s.io/component-base/zpages/flagz/flagreader.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flagz
+
+import (
+	"sort"
+
+	"github.com/spf13/pflag"
+	cliflag "k8s.io/component-base/cli/flag"
+)
+
+type Reader interface {
+	GetFlagz() []Flag
+}
```

Comment:
is there a deeper reason for all these abstractions? Why don't we make it just a `map[string]string` and sort in the handler?

Then `convertNamedFlagSetToFlags` can become a simple `func ConvertNamedFlagSetToFlags(fss) map[string]string`.

Similarly for `type Flag interface`. That's just a pair. Why the interface abstraction and not the type directly?

Reply:
Wiring was happening early, when snapshotting the current state of flags might not be correct (before they are parsed, etc), so making this a concrete `map[string]string` risked grabbing pre-parse values out of flags

Reply:
Ok, I can see that for the reader. Why a Flag abstraction?

Reply:
It was done to limit flagz handler's responsibility to just displaying the passed Flags. And to be able to support flags coming from (possibly) different sources/formats

Reply:
yeah, trying to limit surface area to just name/value instead of taking a hard dep on pflag or NamedFlagSets, etc

Reply:
(passing the underlying type through also made it possible to *mutate* the flags via this wiring, which we *really* didn't want)

Reply:
Anything that speaks against
```golang
type Reader interface {
  GetFlags() map[string]string
}
```
Then I think I am fine with the factoring.

Reply:
I'd be fine with that as well

Reply:
Done